Content.Server/Singularity/EntitySystems/ContainmentFieldGeneratorSystem.cs

```diff
@@ -165,11 +165,13 @@ private void RemoveConnections(Entity<ContainmentFieldGeneratorComponent> genera
             ChangeFieldVisualizer(value.Item1);
         }
         component.Connections.Clear();
-        component.IsConnected = false;
         ChangeOnLightVisualizer(generator);
         ChangeFieldVisualizer(generator);
         _adminLogger.Add(LogType.FieldGeneration, LogImpact.Medium, $"{ToPrettyString(uid)} lost field connections"); // Ideally LogImpact would depend on if there is a singulo nearby
-        _popupSystem.PopupEntity(Loc.GetString("comp-containment-disconnected"), uid, PopupType.LargeCaution);
+
+        if (component.IsConnected)
+            _popupSystem.PopupEntity(Loc.GetString("comp-containment-disconnected"), uid, PopupType.LargeCaution);
+        component.IsConnected = false;
```

Comment:
moving this below ChangeLightVisualizer and ChangeFieldVisualizer breaks shit so just move the popup up instead of moving the boolean set down.

Reply:
done